### DISCUSSION TITLE
Configuring IPv6 Address in a Hook with dhcpcd

### DISCUSSION THREAD

**Discussion by ReeseWang** [rsmarples/dhcpcd#309] [Q&A]
I'm currently working on getting `dhcpcd` to add routes to a routing table that isn't `main`. It appears that the only method to accomplish this is by utilizing the `--noconfigure` command line option, and then setting up the interfaces and routes within hooks.

While implementing this, I logged all the environment variables in my enter hook but couldn't find any variable that holds the IPv6 address provided by the DHCPv6 server. Can someone guide me on how to assign the IPv6 address to my interface in this scenario?

Additionally, is there a `dhcpcd.conf` configuration setting that serves the same purpose as the `--noconfigure` command line option?

**Comment by rsmarples** [comment]
@ReeseWang why don't you want the routes in main? Maybe it's easier to allow dhcpcd to fit your use case?

Here's an example of DHCP6 addresses exported:
```
reason=REBIND6
interface=vioif0
protocol=dhcp6
dhcp6_client_id=000100012bfad097525400737801
dhcp6_server_id=000100062c9f007500bd42090009
dhcp6_ia_na1_iaid=00737801
dhcp6_ia_na1_t1=1000
dhcp6_ia_na1_t2=2000
dhcp6_ia_na1_ia_addr1=fd78::
dhcp6_ia_na1_ia_addr1_pltime=3000
dhcp6_ia_na1_ia_addr1_vltime=4000
dhcp6_name_servers=2001:db8:2::dead:beef 2001:db8:2::cafe:babe
dhcp6_ia_pd1_iaid=00000001
dhcp6_ia_pd1_t1=1000
dhcp6_ia_pd1_t2=2000
dhcp6_ia_pd1_prefix1_pltime=3000
dhcp6_ia_pd1_prefix1_vltime=4000
dhcp6_ia_pd1_prefix1_length=60
dhcp6_ia_pd1_prefix1=2001:db8:8:10::
dhcp6_ia_pd1_prefix1_pd_exclude_prefix_len=77
dhcp6_ia_pd1_prefix1_pd_exclude_subnetID=000800
delegated_dhcp6_prefix=2001:db8:8:8::1/77
```
